slang-checks/src/main/java/com/sonarsource/slang/checks/UnusedFunctionParameterCheck.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * SonarSource SLang
+ * Copyright (C) 2009-2018 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package com.sonarsource.slang.checks;
+
+import com.sonarsource.slang.api.FunctionDeclarationTree;
+import com.sonarsource.slang.api.ParameterTree;
+import com.sonarsource.slang.checks.api.InitContext;
+import com.sonarsource.slang.checks.api.SecondaryLocation;
+import com.sonarsource.slang.checks.api.SlangCheck;
+import com.sonarsource.slang.impl.TopLevelTreeImpl;
+import org.sonar.check.Rule;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static com.sonarsource.slang.utils.SyntacticEquivalence.areEquivalent;
+
+@Rule(key = "S1172")
+public class UnusedFunctionParameterCheck implements SlangCheck {
+
+  @Override
+  public void initialize(InitContext init) {
+    init.register(FunctionDeclarationTree.class, (ctx, functionDeclarationTree) -> {
+      if (!(ctx.parent() instanceof TopLevelTreeImpl)) {
```

Comment:
That's an example of False Negative we might want to track. Only global functions will raise an issue for now. Class methods and lambda will never raise issue for now. You could create a jira ticket related to this